### PR TITLE
Tce support for v3

### PR DESCRIPTION
**Pull request by michelp**

### CODE REVIEW

File: sql/pgsodium--2.0.2--3.0.0.sql [hunk 1]
```diff
@@ -94,18 +89,17 @@ CREATE FUNCTION @extschema@.create_key(
...
-          end,
-          key_context,
-          expires,
-          comment,
-          user_data) RETURNING *
-  $$ LANGUAGE SQL
-  SET search_path='pg_catalog'
-  ;
+  user_data jsonb = null) RETURNS @extschema@.key
+        BEGIN ATOMIC
```
Comment (kiwicopple) [inline sql/pgsodium--2.0.2--3.0.0.sql:93]: `BEGIN ATOMIC`

This implies that the extension only supports Postgres 14+ right?

File: test/tce.sql [hunk 2]
```diff
@@ -37,7 +39,7 @@ SELECT id AS secret2_key_id
 SELECT lives_ok(
   format($test$
          SECURITY LABEL FOR pgsodium ON COLUMN private.bar.secret
-         IS 'ENCRYPT WITH KEY ID %s'
+         IS 'ENCRYPT KEY ID %s ASSOCIATED associated'
```
Comment (kiwicopple) [inline test/tce.sql:42]: Looks like you're removing the "WITH" here, although from a usability perspective I think that adds a lot (assuming I'm reading the functionality correctly: "please encrypt this column with the key XXX using the column `associated`")

File: sql/pgsodium--2.0.2--3.0.0.sql (hunk 1 above)
Comment (michelp) [reply sql/pgsodium--2.0.2--3.0.0.sql:93]: Yeah, I've been going back and forth on that a bit, here's my reasoning:  3.0.0 is not b/w compat, so this is a good time to cut loose older pg versions, esp with 15 right around the corner.  So I think I'll keep the 2.0.x line alive for versions < 14, and I'll backport any fixes to it, but moving forward 3.0 will be 14 and up.

File: test/tce.sql (hunk 2 above)
Comment (michelp) [reply test/tce.sql]: I have restored the `WITH` syntax, it was a bit tedious for testing and I'm usually not a big fan of noise words but I do agree it has some usability to it.

**Pull request merged**